### ISSUE TITLE
Errors thrown by beforeEach leads to child tests left in queue

### ISSUE THREAD

**Issue by sa-spag** [tapjs/tapjs#412]
The following snippet:
```js
const tap = require('tap')

tap.jobs = 1
tap.plan(2)

tap.test('A', async function (t) {
  t.plan(1)
  t.beforeEach(async function () { throw new Error() })
  t.test('a', async function () {})
})

tap.test('B', async function (t) {
  t.plan(1)
  t.test('b', async function () {})
})
```
... outputs:
```
TAP version 13
1..2
# Subtest: A
    1..1
    # Subtest: (anonymous)
        not ok 1 - Error
          ---
          stack: |
            Test.<anonymous> (test.js:8:42)
            Test.<anonymous> (test.js:9:5)
          at:
            line: 8
            column: 42
            file: test.js
            function: Test.<anonymous>
          test: a
          source: |
            t.beforeEach(async function () { throw new Error() })
          ...

        1..1
        # failed 1 test
    not ok 1 - child test left in queue Test: a
    # failed 1 test
not ok 1 - A # time=20.013ms

not ok 2 - child test left in queue Test: B
# failed 2 of 2 tests
# time=30.964ms
```
Indeed, it seems that `process`'s `exit` event is fired after `t.beforeEach` has been processed. Setting `tap.jobs` to `>= 2` badly "solves" the problem as there're are remaining workers to process the test `B`. Note that a `test` or `afterEach` that throws does not have the same effect. Hence my question: is it the intended behavior? Might be related to https://github.com/tapjs/node-tap/issues/212.

**Comment by isaacs** [comment]
Huh!  Yeah, that should be a failure, but not a total abort.  Definitely a bug, thanks for reporting it.

**Comment by usmonster** [comment]
Hello! Just bumping this. The bug's still present, to my knowledge, causing us to use a hacky workaround... I haven't looked deeply at the code, but should this be a straightforward fix? Thanks!

**Comment by isaacs** [comment]
Fixed on v13 5234eda1c8cef2ab23c636e4bbc23a145e9f9ee2
